pulp_container/app/access_policy.py

```diff
@@ -8,9 +8,10 @@ class DistributionAccessPolicyFromDB(AccessPolicyFromDB):
     Access policy for DistributionViewSet which handles namespace permissions.
     """
 
-    def has_add_namespace_perms(self, request, view, action):
+    def has_manage_namespace_distributions_perms(self, request, view, action):
         """
-        Check whether a user can create a namespace or can read the namespace if it already exists.
+        Check whether a user can create a namespace or it can manage distributions
+        for existing namespace.
         """
         namespace = request.data["base_path"].split("/")[0]
```

Comment:
The tests are failing because when you issue a request to retrieve a single distribution, you do not provide `base_path`. Maybe, it would be great if we could differentiate between the object creation and retrieval. Otherwise, just check whether `base_path` was provided and if not, use `view.get_object().namespace`.